### DISCUSSION TITLE
Arduino BLE user recognition (cellphone MAC) to fill user Name field

### DISCUSSION THREAD

**Discussion by reganissu** [pisto/blastic-scale#4] [Ideas]
Exploration Materials

    [ESP32 BLEscan.h utility (implemented with FreeRTOS usage](https://github.com/nkolban/ESP32_BLE_Arduino/blob/master/src/BLEScan.h)

    [ARDUINO R4 BLE library](https://github.com/arduino-libraries/ArduinoBLE/tree/master/src/utility)

    [Elettronica In overview article summary (Italian)](https://ei.futuranet.it/2024/11/05/come-creare-un-sistema-di-rilevamento-ble-con-esp32/)

[BLE arduino ESP32 usage documentation](https://github.com/nkolban/esp32-snippets/tree/master/Documentation)

**Comment by pisto** [comment]
As a low priority task, we might want to make a proof of concept for this. However I am skeptical that this can work. The board supports BLE, however we cannot retrieve the RSSI information, and we cannot discern between far and near devices, as it would be implicit with RFID.

This would also need some kind of application installed on the user's smartphone.

**Reply by reganissu** [reply]
Agreed about low priority and POC.
I think instead RSSI is possible as per BLE.rssi() describet at [https://docs.arduino.cc/libraries/arduinoble/](https://docs.arduino.cc/libraries/arduinoble/ ) 
I also think an app on user's smartphone is not needed if the code can implement recognition of BLE address with BLE.address()

> Query the Bluetooth® address of the Bluetooth® Low Energy device.### Syntax

**Reply by pisto** [reply]
I missed the `BLE.rssi()` method. The bridge seems to implement a [low level transport](https://github.com/arduino/uno-r4-wifi-usb-bridge/blob/main/UNOR4USBBridge/cmds_ble_bridge.h), so I guess it can work but I am really not familiar with this HCI interface
